var/spack/repos/builtin/packages/celeritas/package.py

```diff
@@ -81,13 +81,19 @@ class Celeritas(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("py-breathe", type="build", when="+doc")
     depends_on("py-sphinx", type="build", when="+doc")
 
+    with when("+cuda"):
+        depends_on("thrust")
+    with when("+rocm"):
+        depends_on("rocthrust")
+
     for _std in _cxxstd_values:
         depends_on("geant4 cxxstd=" + _std, when="+geant4 cxxstd=" + _std)
         depends_on("root cxxstd=" + _std, when="+root cxxstd=" + _std)
         depends_on("vecgeom cxxstd=" + _std, when="+vecgeom cxxstd=" + _std)
 
+    depends_on("vecgeom +cuda cuda_arch=none", when="+vecgeom +cuda cuda_arch=none")
```

Comment:
What's the use case for CUDA without an architecture? From `build_systems/cuda.py` it looks like `+cuda cuda_arch=none` may be disallowed someday:
```python
    # FIXME: keep cuda and cuda_arch separate to make usage easier until
    # Spack has depends_on(cuda, when='cuda_arch!=None') or alike
```

Reply:
But I think we can remove this again now. I can't see how it would be needed.